src/Mod/TechDraw/App/DimensionFormatter.cpp

```diff
@@ -321,6 +322,27 @@ QString DimensionFormatter::formatValueToSpec(const double value, const QString&
         // Second, try to cut also decimal dot and zeros, if there are just zeros after it
         formattedValue.replace(QRegularExpression(QStringLiteral("([0-9][0-9]*\\.[0-9]*[1-9])00*$")), QStringLiteral("\\1"));
         formattedValue.replace(QRegularExpression(QStringLiteral("([0-9][0-9]*)\\.0*$")), QStringLiteral("\\1"));
+    }
+    else if (formatSpecifier.contains(QRegularExpression(QStringLiteral("%.*[rR]")))) {
+        // round the value to the given precision
+        QString fs = formatSpecifier;
+        QRegularExpression re(QStringLiteral("%(.*)[rR]"));
+        QRegularExpressionMatch match = re.match(fs);
+        double roundValue = value;
+        if (match.hasMatch()) {
+            QString rounderStr = match.captured(1);
+            double rounder = rounderStr.toDouble();
+            // count rounder decimals
+            int dotIndex = rounderStr.indexOf(QStringLiteral("."));
+            if (dotIndex < 0){
+                fs = QStringLiteral("%.0f");
+            } else {
+                int roundDec = rounderStr.size() - dotIndex - 1;
+                fs = QStringLiteral("%.") + QString::number(roundDec) + QStringLiteral("f");
+            }
+            roundValue = std::ceil(value / rounder) * rounder;
+            formattedValue = QString::asprintf(fs.toStdString().c_str(), roundValue);
+        }
```

Comment:
If no match, we need to return something.

```suggestion
        double roundValue = value;
        if (match.hasMatch()) {
            QString rounderStr = match.captured(1);
            double rounder = rounderStr.toDouble();
            // count rounder decimals
            int dotIndex = rounderStr.indexOf(QStringLiteral("."));
            if (dotIndex < 0){
                fs = QStringLiteral("%.0f");
            } else {
                int roundDec = rounderStr.size() - dotIndex - 1;
                fs = QStringLiteral("%.") + QString::number(roundDec) + QStringLiteral("f");
            }
            roundValue = std::ceil(value / rounder) * rounder;
        }
        formattedValue = QString::asprintf(fs.toStdString().c_str(), roundValue);
```

Reply:
It returns a null string at line 352.
But in fact this case never occurs because the format specifier is already parsed in `DimensionFormatter::getPrefixSuffixSpec` and `DimensionFormatter::isTooSmall`

Reply:
@FlachyJoe if the case cannot occur, then why `if (match.hasMatch())`?

But shouldn't we avoid returning a null string? As far as I can see, we return some kind of value if `%w` can't be parsed. If the other failed parsings return something useful, we should as well with `%r` ;)

Reply:
> if the case cannot occur, then why `if (match.hasMatch())`?

To avoid a crash if someone call this function out of the current process.



> we return some kind of value if `%w` can't be parsed

But nothing for `aefg`: line 347 `if isNumericFormat(formatSpecifier)`.